### PR TITLE
Align Pressibility.win32 to Pressibility.windows

### PR DESCRIPTION
**Pull request by lenahong**
## Description
Align Pressibility.win32 with Pressibility.windows for the following changes:
- Prevent onKeyUp from firing onPress without preceding onKeyDown [PR](https://github.com/microsoft/react-native-windows/pull/10673)
- Enable Right click [PR](https://github.com/microsoft/react-native-windows/pull/9603)

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why
This change is in context with efforts to migrating FURN to use RN Pressable. Win32 version must been aligned with the Windows ver.

### What
This pr addresses the following issues:
- Pressability.win32.js didn't get the check for primary button only fix. 
- when the user pressed space/enter, and then another key while still holding space/enter down.  Where windows will cancel the click, but win32 will continue with it.

## Screenshots
Add any relevant screen captures here from before or after your changes. 

## Testing
If you added tests that prove your changes are effective or that your feature works, add a few sentences here detailing the added test scenarios.

_Optional_: Describe the tests that you ran locally to verify your changes.
 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/react-native-windows/pull/11864&drop=dogfoodAlpha

### CODE REVIEW

File: change/@office-iss-react-native-win32-0156813c-3e7f-479b-b0e6-d832c314b8dd.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "add isDefaultButto check to win32",
```
Comment (PPatBoyd) [inline change/@office-iss-react-native-win32-0156813c-3e7f-479b-b0e6-d832c314b8dd.json:3]: \#nit typo isDefaultButto**n**

File: packages/@office-iss/react-native-win32/src/Libraries/Pressability/Pressability.win32.js [hunk 2]
```diff
@@ -799,6 +805,12 @@ export default class Pressability {
     }
   }
 
+  // [Win32]
+  // $FlowFixMe - button typing
+  _isDefaultPressButton(button): boolean {
```
Comment (rurikoaraki) [inline packages/@office-iss/react-native-win32/src/Libraries/Pressability/Pressability.win32.js:810]: Button should have a type

Comment (rurikoaraki) [reply packages/@office-iss/react-native-win32/src/Libraries/Pressability/Pressability.win32.js:810]: Nvm I see the comment

**Review by rurikoaraki — APPROVED** [review]

**Comment by rurikoaraki** [comment]
You'll need to backport this up to 0.71, FYI

**Review by PPatBoyd — APPROVED** [review]

**Pull request merged**